### PR TITLE
HPCC-30641 Fetch pod logs for ZAP report

### PR DESCRIPTION
**Pull request by wangkx**
Before this PR, WU ZAP report has only one log file which contains the WU log lines from all pods. In this PR, each pod where the WU works on has a log file in the WU ZAP report.

The createWULogFile() is replaced by new readWULogToFiles() which does the following: 1. check whether Remote Log Accessor is available or not. 2. find out the pods where the WU works on. 3. read the log lines into the log files for the pods.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 1]
```diff
@@ -4201,13 +4198,15 @@ void CWsWuFileHelper::createZAPECLQueryArchiveFiles(IConstWorkUnit* cwu, const c
...
 }
 
-void CWsWuFileHelper::createWULogFile(IConstWorkUnit *cwu, WsWuInfo &winfo, const char *path, CWsWuZAPInfoReq & zapLogFilterOptions)
+void CWsWuFileHelper::readWULogToFiles(IConstWorkUnit *cwu, WsWuInfo &winfo, const char *path, CWsWuZAPInfoReq &zapLogFilterOptions)
 {
     if (cwu->getWuidVersion() == 0)
         return;
 
-    LogAccessLogFormat logFormat = zapLogFilterOptions.logFilter.logDataFormat;
+    if (!queryRemoteLogAccessor())
```
Comment (jakesmith) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp:4206]: Does it call this readWULogToFiles function bare-metal?

**Review by jakesmith — COMMENTED** [review]
@wangkx - looks ok afaict, I assume tested.
Have you also tested that bare-metal ZAP report creation still works in bare-metal?

**Comment by wangkx** [comment]
I tested ZAP report creation on both cloud and bare metal. The ZAP reports are created fine.

**Comment by wangkx** [comment]
In bare metal, the ECLWatch does show a warning: "Warning: may not be able to include log file because Remote Log Access plug-in is not available!" Should I change the code (in CWsWorkunitsEx::onWUGetZAPInfo()) to only show the warning if isContainerized()?
```

```

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 1 above)
Comment (jakesmith) [reply esp/services/ws_workunits/ws_workunitsHelpers.cpp:4206]: ok

**Comment by jakesmith** [comment]
> @jakesmith In bare metal, the ECLWatch does show a warning: "Warning: may not be able to include log file because Remote Log Access plug-in is not available!" Should I change the code (in CWsWorkunitsEx::onWUGetZAPInfo()) to only show the warning if isContainerized()?
> 
> ```
> 
> ```

Does it ever try to use the Remote Log Access interface in bare-metal at the moment?

If not, then yes the warning should be suppressed - but if not new, change that in a separate PR.

If it Remote Log Access can be hit in bare-metal - when does it go one route vs the other?

**Comment by wangkx** [comment]
> Does it ever try to use the Remote Log Access interface in bare-metal at the moment?
no

> If not, then yes the warning should be suppressed - but if not new, change that in a separate PR.
It is from https://github.com/hpcc-systems/HPCC-Platform/pull/16957. I will create a separate JIRA/PR.

**Comment by wangkx** [comment]
https://track.hpccsystems.com/browse/HPCC-31122

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good.

**Comment by wangkx** [comment]
@ghalliday please merge this PR.

**Pull request merged**